packages/zero-runtime/README.md

```diff
@@ -249,3 +252,198 @@ const ExtraHeading = styled(Heading)({
   // ... overridden styled
 });
 ```
+
+#### Typing props
+
+If you use TypeScript, add the props typing before the styles to get the type checking:
+
+```tsx
+const Heading = styled('h1')<{ isError?: boolean }>({
+  color: ({ isError }) => (isError ? 'red' : 'black'),
+});
+```
+
+### Theme
+
+Theme lets you reuse the same values, such as colors, spacing, and typography, across your application. It is a plain object of any structure that you can define in your config file.
+
+For example, in Next.js, you can define the theme in the `next.config.js` file:
+
+```js
+const { withZeroPlugin } = require('@mui/zero-next-plugin');
+
+module.exports = withZeroPlugin(
+  {
+    // ...other nextConfig
+  },
+  {
+    theme: {
+      colors: {
+        primary: 'tomato',
+        secondary: 'cyan',
+      },
+      spacing: {
+        unit: 8,
+      },
+      typography: {
+        fontFamily: 'Inter, sans-serif',
+      },
+      // ...more keys and values, it's free style!
+    },
+  },
+);
+```
+
+#### Accesing theme values
+
+A callback can be used with **styled** and **css** APIs to access the theme values:
+
+```js
+const Heading = styled('h1')(({ theme }) => ({
+  color: theme.colors.primary,
+  fontSize: theme.spacing.unit * 4,
+  fontFamily: theme.typography.fontFamily,
+}));
+```
+
+#### CSS variables support
+
+Zero-runtime can generate CSS variables from the theme values when you wrap your theme with `extendTheme` utility. For example, in a `next.config.js` file:
+
+```js
+const { withZeroPlugin, extendTheme } = require('@mui/zero-next-plugin');
+
+module.exports = withZeroPlugin(
+  {
+    // ...other nextConfig
+  },
+  {
+    theme: extendTheme({
+      colors: {
+        primary: 'tomato',
+        secondary: 'cyan',
+      },
+      spacing: {
+        unit: 8,
+      },
+      typography: {
+        fontFamily: 'Inter, sans-serif',
+      },
+    }),
+  },
+);
+```
+
+The `extendTheme` utility will go through the theme and create a `vars` object which represents the tokens that refer to CSS variables.
```

Comment:
If I understand correctly, `extendTheme` is optional.
When should it be used? What are the benefits compared to the plain theme?

Reply:
Thanks for the questions.

extendTheme is optional (I will mention it in the doc)

You would use it if you want to generate CSS variables.

Reply:
> You would use it if you want zero-runtime to generate CSS variables from the theme.

Is there some added value for the user in doing so?

Reply:
It allows you to support multiple colorSchemes/themes and access css variables as JS object in your runtime code, ie, you could also do this in your code -

```js
import theme from '@mui/zero-runtime/theme';

function Component() {
  return <div style={{[theme.vars.palette.primary.main]: 'red'}} />
}
```